test/system/505-networking-pasta.bats

```diff
@@ -410,43 +410,43 @@ function pasta_test_do() {
 
 ### DNS ########################################################################
 
-@test "External resolver, IPv4" {
-    skip_if_no_ipv4 "IPv4 not routable on the host"
-
-    run_podman '?' run --rm --net=pasta $IMAGE nslookup 127.0.0.1
-
-    assert "$output" =~ "1.0.0.127.in-addr.arpa" \
-           "127.0.0.1 not resolved"
-}
-
-@test "External resolver, IPv6" {
-    skip_if_no_ipv6 "IPv6 not routable on the host"
-
-    run_podman '?' run --rm --net=pasta $IMAGE nslookup ::1
-
-    assert "$output" =~ "1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa" \
-           "::1 not resolved"
+@test "Basic nameserver lookup" {
+    run_podman run --rm --net=pasta $IMAGE nslookup 127.0.0.1
```

Comment:
actually looks like we cannot lookup 127.0.0.1 when it is resolved by a public dns server
at least CI seems to failing with it

Reply:
Right, the intention of commit f928cf54aa17 ("pasta: Re-enable "Local forwarder, IPv4" test, accept NXDOMAIN as response") was exactly this: to accept NXDOMAIN as a successful response.

Either we stick to that, or, if the other types of "certain sorts of failure" @dgibson mentions are actually problematic, I guess we'll have to include `dig` in the images: `dig` has two advantages: it exits with zero as long as it gets _any_ dns response (including NXDOMAIN`), and it can be forced to use IPv4 or IPv6.

Reply:
is there any reason to use 127.0.0.1 at all? Why not just `passt.top` or `google.com` as domain name if the only goal is to verify that dns works

Reply:
Because they're not universally resolvable (passt.top couldn't be resolved in some... corporate networks a couple of years ago, because of the bad '.top' reputation, and google.com might not be resolved in some locations).

On top of that, if you just have a local resolver (that is, this test is run without actual connectivity), 127.0.0.1 will actually work with those.

Reply:
See also https://github.com/containers/podman/pull/17713#discussion_r1129604122.

Reply:
There are at least two other tests using google.com in the system tests so clearly everywhere we run these we can use that. This extra 127.0.0.1 dance just doesn't make a ton of sense to me...

Reply:
Well, mind that I'm also running these tests (but not all the system tests) as part of passt's CI, which I would be happy to run without connectivity (one day...).

Reply:
So, I had an earlier draft which removed the lookup of 127.0.0.1, which indeed doesn't work everywhere.  @sbrivio-rh talked me out of it... for now, anyway.  I still think these patches are an improvement over the status quo.

Reply:
Urg, except it breaks the podman CI, because it kinda worked by accident with the test for the string which gets echoed on an NXDOMAIN.

Reply:
it doesn't just break CI, it breaks everywhere where 127.0.0.1 will cause NXDOMAIN which includes local runs depending on your dns server.